File: no5/box.rb
#!/bin/ruby
l = gets.chomp.to_i
n = gets.chomp.to_i
boxes = gets.chomp.split(' ').map! {|box| box.to_i }
boxes.sort!

n.times do | num |
  l -= boxes[num]
  if (l < 0)
    puts num
    break
  elsif (l === 0)
    puts num + 1
    break
  end
end

if l > 0
  puts n
end

File: no5/box.go
package main

import (
  "fmt"
  "sort"
)

func main() {
  var L, N, W int
  fmt.Scan(&L, &N)
  
  
  var boxes []int
  for i:= 0; i < N; i++ {
    fmt.Scan(&W)
    boxes = append(boxes, W)
  }

  sort.Ints(boxes)
  
  for i := 0; i < len(boxes); i++ {
    L -= boxes[i]
    if L == 0 {
      fmt.Println(i + 1)
      break
    } else if L < 0 {
      fmt.Println(i)
      break
    }
  }

  if L > 0 {
    fmt.Println(N)
  }
}